deeppavlov/models/kbqa/query_generator.py

```diff
@@ -0,0 +1,597 @@
+
+        candidate_outputs = []
+
+        if year:
+            candidate_outputs = self.find_relevant_subgraph_cqwn(entity_ids[0][:self.entities_to_leave],
+                                                                 top_rels[:self.rels_to_leave], year)
+        if number:
+            candidate_outputs = self.find_relevant_subgraph_cqwn(entity_ids[0][:self.entities_to_leave],
+                                                                 top_rels[:self.rels_to_leave], number)
+
+        return candidate_outputs
+
+    def complex_question_with_qualifier_solver(self, question: str, entity_ids: List[List[str]]) -> List[Tuple[str]]:
+        ex_rels = []
+        for entity in entity_ids[0][:self.entities_to_leave]:
+            ex_rels += self.wiki_parser("rels", "forw", entity, type_of_rel="direct")
+        ex_rels = list(set(ex_rels))
+        scores = self.rel_ranker([question], ex_rels, False) if self.return_answers else self.rel_ranker(question, ex_rels)
+        top_rels = [score[0] for score in scores]
+        if self.debug:
+            log.debug(f"top scored rels: {top_rels}")
+
+        candidate_outputs = []
+
+        if len(entity_ids) > 1:
+            ent_combs = self.make_entity_combs(entity_ids)
+            candidate_outputs = self.find_relevant_subgraph_cqwq(ent_combs, top_rels[:self.rels_to_leave])
+
+        return candidate_outputs
+
+    def questions_with_count_solver(self, question: str, entity_ids: List[List[str]]) -> List[Tuple[str, str]]:
+        candidate_outputs = []
+
+        ex_rels = []
+        for entity_id in entity_ids:
+            for entity in entity_id[:self.entities_to_leave]:
+                ex_rels += self.wiki_parser("rels", "forw", entity, type_of_rel="direct")
+                ex_rels += self.wiki_parser("rels", "backw", entity, type_of_rel="direct")
+
+        ex_rels = list(set(ex_rels))
+        scores = self.rel_ranker([question], ex_rels, False) if self.return_answers else self.rel_ranker(question, ex_rels)
+        top_rels = [score[0] for score in scores]
+        if self.debug:
+            log.debug(f"top scored rels: {top_rels}")
+        answers = []
+        for entity_id in entity_ids:
+            for entity in entity_id[:self.entities_to_leave]:
+                for rel in top_rels[:self.rels_to_leave]:
+                    answers += self.wiki_parser("objects", "forw", entity, rel, type_of_rel="direct")
+                    if len(answers) > 0:
+                        candidate_outputs.append((rel, str(len(answers))))
+                    else:
+                        answers += self.wiki_parser("objects", "backw", entity, rel, type_of_rel="direct")
+                        candidate_outputs.append((rel, str(len(answers))))
+
+        return candidate_outputs
+
+    def maxmin_one_entity_solver(self, question: str, entities_list: List[str]) -> List[Tuple[str, Any]]:
+        scores = self.rel_ranker([question], self.rank_list_0, False) if self.return_answers \
```

Comment:
`RelRankerInfer.__call__` method receives only two arguments, `False` is unexpected.